src/chronify/time_series_checker.py

```diff
@@ -55,6 +75,31 @@ def _check_expected_timestamps(self) -> int:
         check_timestamp_lists(actual, expected)
         return len(expected)
 
+    def _check_expected_timestamps_with_external_time_zone(self) -> int:
+        assert isinstance(self._time_generator, DatetimeRangeGeneratorExternalTimeZone)
```

Comment:
Is there a way to avoid this special case? Can the special logic be encapsulated within `self._time_range_generator`?